extensions/product-recs/bundle/src/test/java/com/adobe/cq/commerce/extensions/recommendations/internal/servlets/AllowedRecTypesDataSourceServletTest.java

```diff
@@ -0,0 +1,119 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2021 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.commerce.extensions.recommendations.internal.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.ServletException;
+
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.testing.mock.sling.ResourceResolverType;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.adobe.granite.ui.components.Value;
+import com.adobe.granite.ui.components.ds.DataSource;
+import io.wcm.testing.mock.aem.junit.AemContext;
+import io.wcm.testing.mock.aem.junit.AemContextCallback;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class AllowedRecTypesDataSourceServletTest {
+
+    private AllowedRecTypesDataSourceServlet dataSourceServlet;
+
+    @Rule
+    public final AemContext context = createContext("/context/datasource-content.json");
```

Comment:
Please use the `TestContext.newAemContext("/context/datasource-content.json")` factory instead.

Also using context initialiser callbacks is meant to be used by plugins. Tests should use @Before methods instead.

Reply:
Makes sense. The class however is not exported from the core bundle. So I copied it in the extension bundle.